File: src/main/java/sk/pa3kc/entities/Light.java
package sk.pa3kc.entities;

import sk.pa3kc.mylibrary.matrix.pojo.Vector3f;

public class Light {
    private Vector3f position;
    private Vector3f color;

    public Light(Vector3f position, Vector3f color) {
        this.position = position;
        this.color = color;
    }

    public Vector3f getPosition() {
        return this.position;
    }
    public Vector3f getColor() {
        return this.color;
    }
}

File: src/main/kotlin/sk/pa3kc/entity/Light.kt
package sk.pa3kc.entity

import sk.pa3kc.mylibrary.matrix.pojo.Vector3f

data class Light(
    val position: Vector3f,
    val color: Vector3f
) {
    constructor(pX: Float, pY: Float, pZ: Float, cR: Float, cG: Float, cB: Float) : this(
        Vector3f(pX, pY, pZ),
        Vector3f(cR, cG, cB)
    )
}
